ngraph/core/src/op/read_value.cpp

```diff
@@ -97,4 +97,12 @@ bool op::v6::ReadValue::visit_attributes(AttributeVisitor& visitor)
     NGRAPH_OP_SCOPE(v6_ReadValue_visit_attributes);
     visitor.on_attribute("variable_id", m_variable);
     return true;
+}
+
+void op::v6::ReadValue::revalidate_and_infer_types()
```

Comment:
Why do we need to have revalidate for read value?

Reply:
We need to store the state (shape and type) of Variables between inferences (or between evaluate calls), but in some scenarios such as calling the reshape() of CNN Network, we need to drop shape and type and set them to dynamic (shapes, types can be changed, the previous state is invalid)
So we are adding additional functionality to revalidate_and_infer_types() method for ReadValue op to set dynamic state to Variable.

Reply:
`revalidate_and_infer_types` is called for each Node during Function validation which we call after each transformation. is it expected?